tests/Unit/Helpers/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -500,22 +496,79 @@ struct VectorOrArrayAt {
 // given an explicit template parameter pack `Wraps`, wrap the elements of
 // `operand`, passed by pointer, element-by-element, and return the
 // resulting tuple of wrapped elements.
-template <typename... Wraps, typename... Operands, size_t... Is>
+template <typename WrapTuple, typename... Operands, size_t... Is>
 auto wrap_tuple(std::tuple<Operands...>& operand_values,
+                const WrapTuple& wraps,
```

Comment:
can `wraps` be a `std::array<UseRefWrap, sizeof...(Is)>` or a `std::vector`? I wouldn't worry about memory allocations and use whichever is easier.  I think that make the code easier to read (fewer template parameters :) )